tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -83,5 +84,77 @@ private string TranslateError (string culture, string errorCode)
 			PropertyInfo propertyInfo = typeof (MSBStrings).GetProperty (errorCode);
 			return (string) propertyInfo.GetValue (null, null);
 		}
+
+		[TestCase ("cs-CZ")]
+		[TestCase ("de-DE")]
+		[TestCase ("es-ES")]
+		[TestCase ("fr-FR")]
+		[TestCase ("it-IT")]
+		[TestCase ("ja-JP")]
+		[TestCase ("ko-KR")]
+		[TestCase ("pl-PL")]
+		[TestCase ("pt-BR")]
+		[TestCase ("ru-RU")]
+		[TestCase ("tr-TR")]
+		[TestCase ("zh-CN")]
+		[TestCase ("zh-TW")]
+		public void AllErrorTranslation (string culture)
+		{
+			string errorCode = string.Empty;
+			string errorList = string.Empty;
+			CultureInfo originalCulture = Thread.CurrentThread.CurrentUICulture;
+			string path = $"{Directory.GetCurrentDirectory ()}/TaskTests/LocalizationIgnore";
+
+			List<string> commonIgnoreList = ReadFile ($"{path}/common-Translations.ignore");
+			List<string> cultureIgnoreList = ReadFile ($"{path}/{culture}-Translations.ignore");
+
+			foreach (var errorCodeInfo in typeof (MSBStrings).GetProperties ()) {
+				try {
+					errorCode = errorCodeInfo.Name.ToString ();
+					if (errorCode == "ResourceManager" || errorCode == "Culture")
+						continue;
+
+					Assert.IsFalse (string.IsNullOrEmpty (errorCode), "Error code is null or empty");
+					string englishError = TranslateError ("en-US", errorCode);
+					string newCultureError = TranslateError (culture, errorCode);
+
+					if (commonIgnoreList.Contains (errorCode)) {
+						if (englishError != newCultureError)
+							Assert.IsFalse (true, $"{errorCode} is translated. Remove {errorCode} from ~/LocalizationIgnore/common-Translations.ignore");
+					} else if (cultureIgnoreList.Contains (errorCode)) {
+						if (englishError != newCultureError)
+							Assert.IsFalse (true, $"{errorCode} is translated. Remove {errorCode} from ~/LocalizationIgnore/{culture}-Translations.ignore");
+					} else if (englishError == newCultureError)
+						errorList += $"{errorCode} ";
+				} finally {
+					Thread.CurrentThread.CurrentUICulture = originalCulture;
+				}
+			}
+
+			Assert.AreEqual (string.Empty, errorList, $"The following errors were not translated. Add them to ~/LocalizationIgnore/common-Translations.ignore or ~/LocaliztionIgnore/{culture}-Translations.ignore");
+		}
+
+		private List<string> ReadFile (string path)
+		{
+			List<string> ignoreList = new List<string> ();
+			string temp;
+			try {
+				using (var sr = new StreamReader (path)) {
+					while (!sr.EndOfStream){
+						if (sr.Peek () == -1)
+							break;
+						temp = sr.ReadLine ();
+						if (string.IsNullOrEmpty (temp) || temp.Trim ()[0] == '#') {
+							continue;
+						} else {
+							ignoreList.Add (temp.Trim ());
+						}
+					}
+				}
+			} catch (IOException e) {
+				Assert.IsTrue (false, $"Could not read file at: {path}");
+			}
```

Comment:
+1 on @spouliot comment. If you wanted to use a Stream, and for memory reasons go line by line, this is an overkill. You can do the following:
```csharp
using (var reader = new StreamReader (path)) {
	string line;
	while ((line = reader.ReadLine ()) != null) {
		if (line.StartsWith ("#"))
				continue;
		else
				ignoreList.Add (line.Trim ());
	}
}
```

Reply:
Or even a one-liner:

```csharp
return File.ReadAllLines (path).Where (line => !line.StartsWith ("#", StringComparison.Ordinal));
```

Reply:
@rolfbjarne are we playing  ⛳  ?